src/player_avatar.c

```diff
@@ -0,0 +1,731 @@
+        }
+    } else {
+        GF_ASSERT(FALSE);
+        sprite = PlayerAvatar_GetSpriteByStateAndGender(state, gender);
+    }
+    PlayerAvatar_CreateMapObjectWithParams(avatar, manager, sprite, direction, x, z);
+    return avatar;
+}
+
+PlayerAvatar *PlayerAvatar_CreateWithActiveMapObject(MapObjectManager *manager, PlayerSaveData *playerSaveData, u32 gender) {
+    PlayerAvatar *avatar = PlayerAvatar_Create();
+    s32 state = PlayerSaveData_GetState(playerSaveData);
+    PlayerAvatar_Setup(avatar, state, gender, playerSaveData);
+    LocalMapObject *mapObject = PlayerAvatar_GetActiveMapObjectWithMovement1(manager);
+    MapObject_SetSpriteID(mapObject, PlayerAvatar_GetSpriteByStateAndGender(state, gender));
+    MapObject_SetFlagsBits(mapObject, (MapObjectFlagBits)(MAPOBJECTFLAG_UNK13 | MAPOBJECTFLAG_KEEP));
+    MapObject_ClearFlagsBits(mapObject, (MapObjectFlagBits)(MAPOBJECTFLAG_UNK8 | MAPOBJECTFLAG_UNK7));
+    MapObject_SetFlag29(mapObject, TRUE);
+    PlayerAvatar_SetMapObject(avatar, mapObject);
+    return avatar;
+}
+
+void sub_0205C46C(PlayerAvatar *avatar) {
+    LocalMapObject *mapObject = PlayerAvatar_GetMapObject(avatar);
+    GF_ASSERT(mapObject != NULL);
+    MapObject_GetManager(mapObject);
+    ov01_022008B4(avatar);
+    if (PlayerAvatar_GetState(avatar) == PLAYER_STATE_SURFING) {
+        u32 x = PlayerAvatar_GetXCoord(avatar);
+        u32 z = PlayerAvatar_GetZCoord(avatar);
+        u32 direction = PlayerAvatar_GetFacingDirection(avatar);
+        PlayerAvatar_SetUnk34(avatar, ov01_021FE7DC(mapObject, x, z, direction, 1));
+    }
+}
+
+void PlayerAvatar_FreeToHeap(PlayerAvatar *avatar) {
+    Heap_Free(avatar);
+}
+
+void PlayerAvatar_DeleteFromMap(PlayerAvatar *avatar) {
+    MapObject_Remove(PlayerAvatar_GetMapObject(avatar));
+    PlayerAvatar_FreeToHeap(avatar);
+}
+
+static PlayerAvatar *PlayerAvatar_Create(void) {
+    PlayerAvatar *avatar = Heap_Alloc(HEAP_ID_FIELD2, sizeof(PlayerAvatar));
+    GF_ASSERT(avatar != NULL);
+    memset(avatar, 0, sizeof(PlayerAvatar));
+    return avatar;
+}
+
+static void PlayerAvatar_Setup(PlayerAvatar *avatar, s32 state, u32 gender, PlayerSaveData *playerSaveData) {
+    PlayerAvatar_SetPlayerSaveData(avatar, playerSaveData);
+    PlayerAvatar_SetUnk10(avatar, 0);
+    PlayerAvatar_SetUnk14(avatar, 0);
+    PlayerAvatar_SetState(avatar, state);
+    PlayerAvatar_SetGender(avatar, gender);
+    PlayerAvatar_SetTransitionFlags(avatar, 0);
+    PlayerAvatar_ClearUnk24ClearFlag2(avatar);
+    PlayerAvatar_SetUnk28(avatar, -1);
+    PlayerAvatar_SetUnk2C(avatar, -1);
+    PlayerAvatar_SetUnk8(avatar, 255);
+    PlayerAvatar_SetFlag1(avatar, TRUE);
+}
+
+static void PlayerAvatar_CreateMapObjectWithParams(PlayerAvatar *avatar, MapObjectManager *manager, u32 sprite, u32 direction, u32 x, u32 z) {
+    LocalMapObject *mapObject = MapObject_Create(manager, x, z, direction, sprite, 1, 1);
+    GF_ASSERT(mapObject != NULL);
+    MapObject_SetID(mapObject, 255);
```

Comment:
Do we want to have a constant for the player map id? I remember from the comms code that it does actually become relevant for defining new players (in the union room) as PLAYER_MAP_ID + COMM_ID

Reply:
We already have one, `obj_player`